src/server.js

```diff
@@ -39,23 +47,25 @@ class ExecSession {
     this.execStream = await this.exec.start(this.attachOptions);
 
     //handling output
-    this.strout = through((data) => {
+    this.strout = through2((data, enc, cb) => {
       this.sendMessage(msgcode.stdout, new Buffer(data));
+      cb();
     });
 
-    this.strerr = through((data) => {
+    this.strerr = through2((data, enc, cb) => {
       this.sendMessage(msgcode.stderr, new Buffer(data));
+      cb();
     });
```

Comment:
Note the while resume/pause thing is hard to get right... but we can't really stream out big files without it.